### DISCUSSION TITLE
Unencrypted iOS Backup Ignoring `-c compatible`

### DISCUSSION THREAD

**Discussion by ghost** [ReagentX/imessage-exporter#295] [Q&A] [answered]
I can create a backup from my Mac and it correctly converts the -c compatible correctly, but when trying to create a backup from an iOS backup, it pulls the backup in 10 seconds (which is impressive) but ignores the -c compatible. Here's the line I'm running:

 `./imessage-exporter -f html -p ~/iphone_backup_latest/00008030-001445C43E60A02E -a iOS -o backup_export -c compatible`

Thank you!

**Comment by ReagentX** [comment]
What is the diagnostic output for the iOS backup? 

`./imessage-exporter -d -p ~/iphone_backup_latest/00008030-001445C43E60A02E`

**Comment by ghost** [comment]
Here's what it says:

```
iMessage Database Diagnostics

Message diagnostic data:
    Total messages: 55764
    Messages not associated with a chat: 333
Attachment diagnostic data:
    Total attachments: 4814
        Data referenced in table: 11.68 GB
        Data present on disk: 9.73 GB
    Missing files: 946 (20%)
        No path provided: 946
        No file located: 0
Global diagnostic data:
    Total database size: 8.25 KB
    Duplicated contacts: 23
    Duplicated chats: 20
Done!
```


I'm working on trying to figure out where the missing attachments are. Potentially not fully pulled from iCloud yet.

**Comment by ReagentX** [comment]
It still detects `9.73 GB` of attachment data, but given your original post it seems none of them are getting copied. Are any errors emitted when you run the export? What version of iOS was the backup made from? What is the full output when you ran your export? 

I just ran an export on an unencrypted backup and it worked fine.

**Comment by ghost** [comment]
The output didn’t give any errors and was about 9.73GB. It just still included HEIF files instead of being converted to JPEG. Basically, it made a direct copy of everything. This is 12.7.5.

**Reply by ReagentX** [reply]
What type of files are not converted? Images end with a HEIC extension, and those are supported. I don't see a HEIF extension in any of my exports. The code currently only supports HEIC:

https://github.com/ReagentX/imessage-exporter/blob/247e01e32e92171bf3d2ccd79e129dd39ea21145/imessage-exporter/src/app/attachment_manager.rs#L152-L162

**Comment by ReagentX** [comment]
Ok, this sounds like a bug. I will investigate.

**Reply by ReagentX** [reply]
Tracking this in #296.

**Comment by ghost** [comment]
Sorry phone typo. Yes, HEIC. Basically nothing is converted, it is just copying the files directly.

**Reply by ReagentX** [reply]
I think I know what this is; on iOS the attachment files are stored with names as hashes without extensions, so the current detection code doesn't work in those cases. This will get fixed in #296. 


https://github.com/ReagentX/imessage-exporter/blob/247e01e32e92171bf3d2ccd79e129dd39ea21145/imessage-database/src/tables/attachment.rs#L410-L420

**Reply by ReagentX** [reply, marked as answer]
This was resolved #297 and will be released in California Poppy (1.10).

**Comment by ghost** [comment]
Great! Just trying to be helpful. 

Sorry for the small file size. 10mb limit.

Here's the process. The folder is full of HEIC files. But it couldn't even possibly be converting it that fast on this i5 Mac. To convert from the Mac iMessage program, it takes about 5 minutes to do this. 

https://github.com/user-attachments/assets/04160566-75a2-470c-b322-3a3eaeb65ae8

**Reply by ReagentX** [reply]
Thanks, this is helpful.
